packages/storage-plus/src/prefix.rs

```diff
@@ -99,21 +112,26 @@ where
     }
 }
 
-impl<T> Prefix<T>
+impl<K, T> Prefix<K, T>
 where
+    K: KeyDeserialize,
     T: Serialize + DeserializeOwned,
 {
     pub fn new(top_name: &[u8], sub_names: &[&[u8]]) -> Self {
-        Prefix::with_deserialization_function(top_name, sub_names, &[], default_deserializer)
+        Prefix::with_deserialization_function(
+            top_name,
+            sub_names,
+            &[],
+            default_deserializer_kv::<K, T>,
```

Comment:
Does this actually solve the custom deserialiser issue for multi-index?

Reply:
I would address that in a follow-up PR, yes.